packages/protvista-links/src/links-parser.ts

```diff
@@ -0,0 +1,81 @@
+const parseToRowData = (text: string, threshold: number): ArrayOfNumberArray =>
+  text
+    .split("\n")
+    .slice(1)
+    .map((line) => {
+      const [n1, n2, _, __, p] = line.split(" ");
+      return [+n1, +n2, +p];
+    })
+    .filter(([_, __, p]) => p > threshold);
+
+export const parseLinksAssociative = (
+  text: string,
+  threshold: number
+): LinksObject => {
+  const rawData = parseToRowData(text, threshold);
+  const n2set: NumberArray = {};
+  const sets: ArrayOfNumberArray = [];
+  rawData.forEach(([n1, n2]) => {
+    if (!n2set[n1] && !n2set[n2]) {
+      n2set[n1] = sets.length;
+      n2set[n2] = sets.length;
+      sets.push([n1, n2]);
+    } else if (n2set[n1] && !n2set[n2]) {
+      n2set[n2] = n2set[n1];
+      sets[n2set[n1]].push(n2);
+    } else if (!n2set[n1] && n2set[n2]) {
+      n2set[n1] = n2set[n2];
+      sets[n2set[n1]].push(n1);
+    } else if (n2set[n1] && n2set[n2] && n2set[n1] !== n2set[n2]) {
+      sets[n2set[n2]].forEach((n) => {
+        sets[n2set[n1]].push(n);
+        n2set[n] = n2set[n1];
+      });
+    }
```

Comment:
I don't understand the context completely so this question might not make sense: do you ever care about the case where `n2set[n1] && n2set[n2] && n2set[n1] === n2set[n2]`?

Reply:
So the `parseLinksAssociative` idea is that if position 20 is linked with 30, and position 30 is linked with 40, then I can infer that position 20 is linked with 40. Which I thought was the logic of the contacts I need to display, but it is not, a link has to be explicitly included. So I'm not using this function right now, but I though of leaving it there in case there is a future case for it.

in any case, this is logic to ignore the case `n2set[n1] && n2set[n2] && n2set[n1] === n2set[n2]`:
* if `n2set[n1]` exists it implies that `n1` belongs to that set
* if `n2set[n2]` exists it implies that `n2` belongs to that set
* if `n2set[n1] === n2set[n2]`, they are the same set, so I don't need to include anything in the set because they are already there, and any associative link has already taken care of.